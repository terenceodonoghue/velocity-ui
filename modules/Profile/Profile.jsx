/** @jsx jsx */
import { jsx } from '@emotion/core';
import { string } from 'prop-types';
import { Avatar, Button, Container } from 'components';
import * as S from './Profile.styles';

const Sidebar = ({
  avatarSrc, bio, email, location, name, phone, title,
}) => (
  <S.Sidebar>
    <Container.Card>
      <S.Avatar>
        <Avatar.Round src={avatarSrc} height="140" width="140" />
      </S.Avatar>
      <S.Name>{name}</S.Name>
      <S.Title>{title}</S.Title>
      <S.Buttons>
        <S.Button>
          <Button.Contained width="100%">Edit profile</Button.Contained>
        </S.Button>
        <S.Button>
          <Button.Outlined width="100%">Change status</Button.Outlined>
        </S.Button>
      </S.Buttons>
      <S.Rule />
      <S.DetailsList>
        <S.DetailsTerm>Role</S.DetailsTerm>
        <S.DetailsDescription>Administrator, Moderator</S.DetailsDescription>
        <S.DetailsTerm>E-mail</S.DetailsTerm>
        <S.DetailsDescription>{email}</S.DetailsDescription>
        <S.DetailsTerm>Phone</S.DetailsTerm>
        <S.DetailsDescription>{phone}</S.DetailsDescription>
        <S.DetailsTerm>Twitter</S.DetailsTerm>
        <S.DetailsDescription>@invisionapp</S.DetailsDescription>
        <S.DetailsTerm>Location</S.DetailsTerm>
        <S.DetailsDescription>{location}</S.DetailsDescription>
        <S.DetailsTerm>Bio</S.DetailsTerm>
        <S.DetailsDescription>{bio}</S.DetailsDescription>
      </S.DetailsList>
    </Container.Card>
  </S.Sidebar>
);

Sidebar.propTypes = {
  avatarSrc: string.isRequired,
  bio: string.isRequired,
  email: string.isRequired,
  location: string.isRequired,
  name: string.isRequired,
  phone: string.isRequired,
  title: string.isRequired,
};

export default {
  Sidebar,
};
