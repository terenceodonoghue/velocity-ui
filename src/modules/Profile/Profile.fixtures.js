/* eslint-disable import/prefer-default-export */
import faker from 'faker';

export const sidebar = {
  avatarSrc: faker.image.avatar(),
  bio:
    "This theme adds additional client and channel information. It's useful for Admins when you need any ID of a user without searching for it",
  email: 'invision@invisionapp.com',
  location: 'New York, NY',
  name: 'Anna Black',
  phone: '+144-3412-4422',
  roles: 'Administrator, Moderator',
  title: 'Sr. Customer Manager',
};
