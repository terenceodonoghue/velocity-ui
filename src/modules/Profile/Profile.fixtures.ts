/* eslint-disable import/prefer-default-export */
import * as faker from 'faker';

export const sidebar = {
  bio:
    "This theme adds additional client and channel information. It's useful for Admins when you need any ID of a user without searching for it",
  email: 'invision@invisionapp.com',
  jobTitle: 'Sr. Customer Manager',
  location: 'New York, NY',
  name: 'Anna Black',
  phone: '+144-3412-4422',
  roles: 'Administrator, Moderator',
  src: faker.image.avatar(),
};
