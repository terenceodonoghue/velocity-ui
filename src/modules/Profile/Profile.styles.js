import styled from '@emotion/styled';
import { Avatar, Button, Container } from '../../components';

export const Buttons = styled.div`
  display: flex;
  margin: 0 -8px 40px;
`;

export const EditButton = styled(Button.Contained)`
  flex: 1;
  margin: 0 8px;
  width: unset;
`;

export const Label = styled.dt`
  color: #b0bac9;
  margin-bottom: 10px;
  font-size: 12px;
  letter-spacing: 1.1px;
  text-transform: uppercase;
`;

export const Name = styled.h2`
  margin: 0 0 3px;
  font-size: 28px;
  font-weight: var(--font-weight-light);
  text-align: center;
`;

export const ProfilePicture = styled(Avatar.Round)`
  display: block;
  height: 140px;
  margin: 32px auto 19px;
  width: 140px;
`;

export const Rule = styled.hr`
  border: 0;
  height: 1px;
  margin-bottom: 42px;
  background-color: #e4e8f0;
`;

export const Sidebar = styled(Container.Card)`
  height: 100%;
  width: 357px;
`;

export const StatusButton = styled(Button.Outlined)`
  flex: 1;
  margin: 0 8px;
  width: unset;
`;

export const Title = styled.h3`
  margin: 0 0 33px;
  color: var(--color-bluey-grey);
  font-size: var(--base-font-size);
  font-weight: var(--font-weight-regular);
  text-align: center;
`;

export const Value = styled.dd`
  margin: 0 0 24px;
  line-height: 1.47;
`;
