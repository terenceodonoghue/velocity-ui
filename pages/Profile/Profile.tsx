/* eslint-disable import/prefer-default-export */
/** @jsx jsx */
import { jsx } from '@emotion/core';
import faker from 'faker';
import { FunctionComponent } from 'react';
import { Avatar, Button, Layout } from '~/components';
import * as css from './Profile.styles';

interface SidebarProps {
  bio: string;
  email: string;
  jobTitle: string;
  location: string;
  name: string;
  phone: string;
  roles: string;
  src: string;
}

export const Sidebar: FunctionComponent<SidebarProps> = ({
  bio,
  email,
  jobTitle,
  location,
  name,
  phone,
  roles,
  src,
}) => (
  <Layout.Card css={css.sidebar}>
    <Avatar
      alt={faker.name.findName()}
      css={css.sidebarAvatar}
      round
      src={src}
    />
    <h2 css={css.sidebarName}>{name}</h2>
    <h3 css={css.sidebarJobTitle}>{jobTitle}</h3>
    <div css={css.sidebarButtons}>
      <Button.Contained css={css.sidebarButton}>Edit profile</Button.Contained>
      <Button.Outlined css={css.sidebarButton}>Change status</Button.Outlined>
    </div>
    <hr css={css.sidebarDivider} />
    <dl>
      <dt css={css.sidebarFieldLabel}>Role</dt>
      <dd css={css.sidebarFieldValue}>{roles}</dd>
      <dt css={css.sidebarFieldLabel}>E-mail</dt>
      <dd css={css.sidebarFieldValue}>{email}</dd>
      <dt css={css.sidebarFieldLabel}>Phone</dt>
      <dd css={css.sidebarFieldValue}>{phone}</dd>
      <dt css={css.sidebarFieldLabel}>Twitter</dt>
      <dd css={css.sidebarFieldValue}>@invisionapp</dd>
      <dt css={css.sidebarFieldLabel}>Location</dt>
      <dd css={css.sidebarFieldValue}>{location}</dd>
      <dt css={css.sidebarFieldLabel}>Bio</dt>
      <dd css={css.sidebarFieldValue}>{bio}</dd>
    </dl>
  </Layout.Card>
);
