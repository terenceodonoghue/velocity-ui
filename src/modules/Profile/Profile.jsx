import React from 'react';
import { string } from 'prop-types';
import * as S from './Profile.styles';

const Sidebar = ({
  avatarSrc, bio, email, location, name, phone, roles, title,
}) => (
  <S.Sidebar>
    <S.ProfilePicture src={avatarSrc} />
    <S.Name>{name}</S.Name>
    <S.Title>{title}</S.Title>
    <S.Buttons>
      <S.EditButton>
        Edit profile
      </S.EditButton>
      <S.StatusButton>
        Change status
      </S.StatusButton>
    </S.Buttons>
    <S.Rule />
    <dl>
      <S.Label>Role</S.Label>
      <S.Value>{roles}</S.Value>
      <S.Label>E-mail</S.Label>
      <S.Value>{email}</S.Value>
      <S.Label>Phone</S.Label>
      <S.Value>{phone}</S.Value>
      <S.Label>Twitter</S.Label>
      <S.Value>@invisionapp</S.Value>
      <S.Label>Location</S.Label>
      <S.Value>{location}</S.Value>
      <S.Label>Bio</S.Label>
      <S.Value>{bio}</S.Value>
    </dl>
  </S.Sidebar>
);

Sidebar.propTypes = {
  avatarSrc: string.isRequired,
  bio: string.isRequired,
  email: string.isRequired,
  location: string.isRequired,
  name: string.isRequired,
  phone: string.isRequired,
  roles: string.isRequired,
  title: string.isRequired,
};

export default {
  Sidebar,
};
