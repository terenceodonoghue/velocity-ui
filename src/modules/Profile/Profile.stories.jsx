import React from 'react';
import { storiesOf } from '@storybook/react';
import Profile from './Profile';
import { sidebar } from './Profile.fixtures';

storiesOf('Modules|User Profile', module).add('Sidebar', () => (
  <Profile.Sidebar {...sidebar} />
));
