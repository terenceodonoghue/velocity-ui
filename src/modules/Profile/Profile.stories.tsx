import React from 'react';
import { storiesOf } from '@storybook/react';
import { Sidebar } from './Profile';
import * as fixtures from './Profile.fixtures';

storiesOf('Modules/User Profile', module).add('Sidebar', () => (
  <Sidebar {...fixtures.sidebar} />
));
