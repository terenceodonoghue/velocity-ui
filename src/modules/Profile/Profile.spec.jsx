import React from 'react';
import { shallow } from 'enzyme';
import faker from 'faker';
import Profile from './Profile';
import * as S from './Profile.styles';

describe('Profile', () => {
  describe('Sidebar', () => {
    const props = {
      bio: faker.lorem.paragraph(),
      email: faker.internet.email(),
      location: `${faker.address.city()}, ${faker.address.stateAbbr()}`,
      name: faker.name.findName(),
      phone: faker.phone.phoneNumberFormat(2),
      title: 'Sr. Customer Manager',
      roles: 'Administrator, Moderator',
      srcUrl: faker.image.avatar(),
    };

    it('should render bio', () => {
      const component = shallow(
        <Profile.Sidebar {...props} />,
      );

      expect(
        component.contains(<S.DetailsTerm>Bio</S.DetailsTerm>)
        && component.contains(<S.DetailsDescription>{props.bio}</S.DetailsDescription>),
      ).toBe(true);
    });

    it('should render email', () => {
      const component = shallow(
        <Profile.Sidebar {...props} />,
      );

      expect(
        component.contains(<S.DetailsTerm>E-mail</S.DetailsTerm>)
        && component.contains(<S.DetailsDescription>{props.email}</S.DetailsDescription>),
      ).toBe(true);
    });

    it('should render location', () => {
      const component = shallow(
        <Profile.Sidebar {...props} />,
      );

      expect(
        component.contains(<S.DetailsTerm>Location</S.DetailsTerm>)
        && component.contains(<S.DetailsDescription>{props.location}</S.DetailsDescription>),
      ).toBe(true);
    });

    it('should render name', () => {
      const component = shallow(
        <Profile.Sidebar {...props} />,
      );

      expect(
        component.contains(<S.Name>{props.name}</S.Name>),
      ).toBe(true);
    });

    it('should render phone', () => {
      const component = shallow(
        <Profile.Sidebar {...props} />,
      );

      expect(
        component.contains(<S.DetailsTerm>Phone</S.DetailsTerm>)
        && component.contains(<S.DetailsDescription>{props.phone}</S.DetailsDescription>),
      ).toBe(true);
    });

    it('should render title', () => {
      const component = shallow(
        <Profile.Sidebar {...props} />,
      );

      expect(
        component.contains(<S.Title>{props.title}</S.Title>),
      ).toBe(true);
    });

    it('should render roles', () => {
      const component = shallow(
        <Profile.Sidebar {...props} />,
      );

      expect(
        component.contains(<S.DetailsTerm>Role</S.DetailsTerm>)
        && component.contains(<S.DetailsDescription>{props.roles}</S.DetailsDescription>),
      ).toBe(true);
    });

    it('should render profile picture', () => {
      const component = shallow(
        <Profile.Sidebar {...props} />,
      );

      expect(
        component.contains(<S.ProfilePicture src={props.srcUrl} />),
      ).toBe(true);
    });
  });
});
