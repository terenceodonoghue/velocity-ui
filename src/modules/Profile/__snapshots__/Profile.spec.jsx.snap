// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Sidebar matches snapshot 1`] = `
.emotion-42 {
  display: block;
  border-radius: 1px;
  border: solid 1px rgba(46,91,255,0.08);
  box-shadow: 0 10px 20px 0 rgba(46,91,255,0.07);
  padding: 32px 24px;
  background-color: var(--color-white);
  height: 100%;
  width: 357px;
}

.emotion-1 {
  border-radius: 50%;
  display: block;
  height: 140px;
  margin: 32px auto 19px;
  width: 140px;
}

.emotion-3 {
  margin: 0 0 3px;
  font-size: 28px;
  font-weight: var(--font-weight-light);
  text-align: center;
}

.emotion-5 {
  margin: 0 0 33px;
  color: var(--color-bluey-grey);
  font-size: var(--base-font-size);
  font-weight: var(--font-weight-regular);
  text-align: center;
}

.emotion-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 -8px 40px;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  padding: 0;
  font-size: var(--base-font-size);
  font-weight: var(--font-weight-medium);
  cursor: pointer;
  -webkit-transition: var(--transition);
  transition: var(--transition);
  border-radius: 4px;
  height: 40px;
  width: 200px;
  background-color: var(--color-clear-blue);
  color: white;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: 0 8px;
  width: unset;
}

.emotion-8:disabled {
  opacity: 0.3;
  pointer-events: none;
}

.emotion-8:focus {
  outline: 0;
}

.emotion-8:enabled:hover {
  background-color: var(--color-clear-blue-dark);
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  padding: 0;
  font-size: var(--base-font-size);
  font-weight: var(--font-weight-medium);
  cursor: pointer;
  -webkit-transition: var(--transition);
  transition: var(--transition);
  border-radius: 4px;
  height: 40px;
  width: 200px;
  background: rgba(46,91,255,0.2);
  color: var(--color-clear-blue);
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: 0 8px;
  width: unset;
}

.emotion-11:disabled {
  opacity: 0.3;
  pointer-events: none;
}

.emotion-11:focus {
  outline: 0;
}

.emotion-11:enabled:hover {
  background-color: rgba(46,91,255,0.3);
}

.emotion-15 {
  border: 0;
  height: 1px;
  margin-bottom: 42px;
  background-color: #e4e8f0;
}

.emotion-17 {
  color: #b0bac9;
  margin-bottom: 10px;
  font-size: 12px;
  -webkit-letter-spacing: 1.1px;
  -moz-letter-spacing: 1.1px;
  -ms-letter-spacing: 1.1px;
  letter-spacing: 1.1px;
  text-transform: uppercase;
}

.emotion-19 {
  margin: 0 0 24px;
  line-height: 1.47;
}

<div
  className="emotion-41 emotion-42 emotion-43"
>
  <img
    className="emotion-0 emotion-1 emotion-2"
    src="./users/joan.png"
  />
  <h2
    className="emotion-3 emotion-4"
  >
    Anna Black
  </h2>
  <h3
    className="emotion-5 emotion-6"
  >
    Sr. Customer Manager
  </h3>
  <div
    className="emotion-13 emotion-14"
  >
    <button
      className="emotion-7 emotion-8 emotion-9"
    >
      Edit profile
    </button>
    <button
      className="emotion-10 emotion-11 emotion-12"
    >
      Change status
    </button>
  </div>
  <hr
    className="emotion-15 emotion-16"
  />
  <dl>
    <dt
      className="emotion-17 emotion-18"
    >
      Role
    </dt>
    <dd
      className="emotion-19 emotion-20"
    >
      Administrator, Moderator
    </dd>
    <dt
      className="emotion-17 emotion-18"
    >
      E-mail
    </dt>
    <dd
      className="emotion-19 emotion-20"
    >
      invision@invisionapp.com
    </dd>
    <dt
      className="emotion-17 emotion-18"
    >
      Phone
    </dt>
    <dd
      className="emotion-19 emotion-20"
    >
      +144-3412-4422
    </dd>
    <dt
      className="emotion-17 emotion-18"
    >
      Twitter
    </dt>
    <dd
      className="emotion-19 emotion-20"
    >
      @invisionapp
    </dd>
    <dt
      className="emotion-17 emotion-18"
    >
      Location
    </dt>
    <dd
      className="emotion-19 emotion-20"
    >
      New York, NY
    </dd>
    <dt
      className="emotion-17 emotion-18"
    >
      Bio
    </dt>
    <dd
      className="emotion-19 emotion-20"
    >
      This theme adds additional client and channel information. It's useful for Admins when you need any ID of a user without searching for it
    </dd>
  </dl>
</div>
`;
